src/common/webrequest.cpp

```diff
@@ -0,0 +1,485 @@
+    m_bytesReceived += sizeReceived;
+}
+
+// The SplitParamaters implementation is adapted to wxWidgets
+// from Poco::Net::MessageHeader::splitParameters
+
+void wxWebRequest::SplitParameters(const wxString& s, wxString& value,
+    wxWebRequestHeaderMap& parameters)
+{
+    value.clear();
+    parameters.clear();
+    wxString::const_iterator it = s.begin();
+    wxString::const_iterator end = s.end();
+    while ( it != end && wxIsspace(*it) ) ++it;
+    while ( it != end && *it != ';' ) value += *it++;
+    value.Trim();
+    if ( it != end ) ++it;
+    SplitParameters(it, end, parameters);
+}
+
+void wxWebRequest::SplitParameters(const wxString::const_iterator& begin,
+    const wxString::const_iterator& end, wxWebRequestHeaderMap& parameters)
+{
+    wxString pname;
+    wxString pvalue;
+    pname.reserve(32);
+    pvalue.reserve(64);
+    wxString::const_iterator it = begin;
+    while ( it != end )
+    {
+        pname.clear();
+        pvalue.clear();
+        while ( it != end && wxIsspace(*it) ) ++it;
+        while ( it != end && *it != '=' && *it != ';' ) pname += *it++;
+        pname.Trim();
+        if ( it != end && *it != ';' ) ++it;
+        while ( it != end && wxIsspace(*it) ) ++it;
+        while ( it != end && *it != ';' )
+        {
+            if ( *it == '"' )
+            {
+                ++it;
+                while ( it != end && *it != '"' )
+                {
+                    if ( *it == '\\' )
+                    {
+                        ++it;
+                        if ( it != end ) pvalue += *it++;
+                    }
+                    else pvalue += *it++;
+                }
+                if ( it != end ) ++it;
+            }
+            else if ( *it == '\\' )
+            {
+                ++it;
+                if ( it != end ) pvalue += *it++;
+            }
+            else pvalue += *it++;
+        }
+        pvalue.Trim();
+        if ( !pname.empty() ) parameters[pname] = pvalue;
+        if ( it != end ) ++it;
+    }
+}
+
+void wxWebRequest::ProcessStateEvent(State state, const wxString& failMsg)
+{
+    if (!IsActiveState(state) && GetResponse())
+        GetResponse()->Finalize();
+
+    wxString responseFileName;
+
+    wxWebRequestEvent evt(wxEVT_WEBREQUEST_STATE, GetId(), state,
+        GetResponse(), failMsg);
+    if ( state == State_Completed && m_storage == Storage_File )
+    {
+        responseFileName = GetResponse()->GetFileName();
+        evt.SetResponseFileName(responseFileName);
+    }
+
+    ProcessEvent(evt);
+
+    // Remove temporary file if it still exists
+    if ( state == State_Completed && m_storage == Storage_File &&
+        wxFileExists(responseFileName) )
+        wxRemoveFile(responseFileName);
+
+    // Remove reference after the request is no longer active
+    if ( !IsActiveState(state) )
+        DecRef();
+}
+
+//
+// wxWebResponse
+//
+wxWebResponse::wxWebResponse(wxWebRequest& request) :
+    m_request(request),
+    m_readSize(8 * 1024)
```

Comment:
This looks tiny... I'd use 64KiB if I had to choose arbitrarily, although ideal would be to benchmark it with a few different buffer sizes.